docs/design/adrs/2024-04-30-remove-langchain-from-tools.md

```diff
@@ -0,0 +1,87 @@
+# Remove Langchain from tools
+
+* **Status:** proposed
+* **Proposer:**  @adamdougal
+* **Date:** 2024-04-30
+* **Technical Story:** [Make required changes to store image embeddings](https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/issues/748)
+
+## Context and Problem Statement
+
+In order to enhance our data querying capabilities to include image-based data, we need to incorporate image embeddings
+generated by [Azure Computer Vision](https://learn.microsoft.com/en-us/azure/ai-services/computer-vision/how-to/image-retrieval?tabs=csharp#call-the-vectorize-image-api).
+Furthermore, we aim to generate image captions using `gpt-4-vision` and store the embeddings of these captions alongside
+the image embeddings. This necessitates the introduction of an **additional** field in the index to accommodate multiple
+embedding vectors, as demonstrated in the [spike notebook](../../spikes/using-image-data/ai-vision.ipynb).
+
+Currently, we utilise [Langchain](https://python.langchain.com/docs/get_started/quickstart/) to create and query the
+Azure AI Search index. However, as of the time of writing, Langchain does not support querying Azure AI Search for
+multiple embedding vectors in a single query. Additionally, it lacks support for using Azure Computer Vision to generate
+embeddings, as can be seen in the [list of supported embedding models](https://python.langchain.com/docs/integrations/text_embedding/).
+
+The purpose of this ADR is to determine the best approach to incorporate the ability to query image-based data, given
+the aforementioned constraints.
+
+## Decision Drivers
+
+* Certainty of delivery
+* Speed of delivery
+* Zero loss of existing functionality
+* Code complexity
+* Maintainability
+
+## Considered Options
+
+* Remove Langchain from tools
+* Contribute features back to Langchain
+* Keep Langchain but not if advanced image processing enabled
+
+## Decision Outcome
+
+Remove Langchain from tools. This gives us full control to deliver this feature. It will also allow us to implement
+cutting edge features in future without having to wait for Langchain to be updated.
+
+## Pros and Cons of the Options
+
+### Remove Langchain from tools
```

Comment:
This section is missing details on what we would lose out on if we removed langchain from "tools".
It would be beneficial to understand why the original contributors chose to include langchain in non-orchestration code.
It would also make sense to add a brief summary of what these tools are doing - is it the search flow? How is langchain contributing to it currently?

Reply:
Good point! I've added a note in the bad section for this.